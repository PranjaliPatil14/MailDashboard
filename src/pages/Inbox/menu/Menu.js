import React from "react";
import * as PropTypes from "prop-types";
import Button from "../../../components/button/Button";
import Folders from "./Folders";
import Categories from "./Categories";
import Labels from "./Labels";
import "./menu.css";

const Menu = ({ setShowCompose }) => {
  return (
    <div className="inbox-menu">
      <Button
        type="submit"
        onClick={() => setShowCompose(true)}
        className="compose-mail"
      >
        Compose Mail
      </Button>
      <Folders />
      <Categories />
      <Labels />
    </div>
  );
};

Menu.propTypes = {
  setShowCompose: PropTypes.func.isRequired,
};

export default Menu;
