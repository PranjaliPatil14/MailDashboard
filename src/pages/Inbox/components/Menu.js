import React from "react";
import Button from "../../../components/button/Button";
import "./menu.css";
import Folders from "./Folders";
import Categories from "./Categories";
import Labels from "./Labels";

const Menu = () => {
  return (
    <div className="inbox-menu">
      <Button type="submit" onClick={() => {}} className="compose-mail">
        Compose Mail
      </Button>
      <Folders />
      <Categories />
      <Labels />
    </div>
  );
};

export default Menu;
