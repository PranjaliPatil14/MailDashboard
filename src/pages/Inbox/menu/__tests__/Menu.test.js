import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Menu from "../Menu";

describe("Menu", () => {
  test("should render inbox menu", () => {
    const setShowCompose = jest.fn();
    const { asFragment, getByText } = render(
      <Menu setShowCompose={setShowCompose} />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText("Compose Mail"));

    expect(setShowCompose).toHaveBeenCalledWith(true);
  });
});
