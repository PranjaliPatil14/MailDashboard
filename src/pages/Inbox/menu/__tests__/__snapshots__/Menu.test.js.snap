// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render inbox menu 1`] = `
<DocumentFragment>
  <div
    class="inbox-menu"
  >
    <button
      class="button compose-mail"
      data-testid=""
      type="submit"
    >
      Compose Mail
    </button>
    <div
      class="menu-folders"
    >
      <p
        class="menu-folders-title"
      >
        Folders
      </p>
      <ul
        class="folders"
      >
        <li
          class="folder-item"
        >
          <svg
            class="folder-item-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="22 12 16 12 14 15 10 15 8 12 2 12"
            />
            <path
              d="M5.45 5.11L2 12v6a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2v-6l-3.45-6.89A2 2 0 0 0 16.76 4H7.24a2 2 0 0 0-1.79 1.11z"
            />
          </svg>
           Inbox
        </li>
        <li
          class="folder-item"
        >
          <svg
            class="folder-item-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
            />
            <polyline
              points="22,6 12,13 2,6"
            />
          </svg>
           Sent Mails
        </li>
        <li
          class="folder-item"
        >
          <svg
            class="folder-item-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
          </svg>
           Important
        </li>
        <li
          class="folder-item"
        >
          <svg
            class="folder-item-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 19.5A2.5 2.5 0 0 1 6.5 17H20"
            />
            <path
              d="M6.5 2H20v20H6.5A2.5 2.5 0 0 1 4 19.5v-15A2.5 2.5 0 0 1 6.5 2z"
            />
          </svg>
           Drafts
        </li>
        <li
          class="folder-item"
        >
          <svg
            class="folder-item-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="3 6 5 6 21 6"
            />
            <path
              d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
            />
            <line
              x1="10"
              x2="10"
              y1="11"
              y2="17"
            />
            <line
              x1="14"
              x2="14"
              y1="11"
              y2="17"
            />
          </svg>
           Trash
        </li>
      </ul>
    </div>
    <div
      class="menu-categories"
    >
      <p
        class="menu-categories-title"
      >
        Categories
      </p>
      <ul
        class="categories"
      >
        <li
          class="category-item"
        >
          <svg
            class="category-item-icon"
            fill="green"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
          </svg>
           Work
        </li>
        <li
          class="category-item"
        >
          <svg
            class="category-item-icon"
            fill="red"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
          </svg>
           Document
        </li>
        <li
          class="category-item"
        >
          <svg
            class="category-item-icon"
            fill="yellow"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
          </svg>
           Social
        </li>
        <li
          class="category-item"
        >
          <svg
            class="category-item-icon"
            fill="orange"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
          </svg>
           Advertising
        </li>
        <li
          class="category-item"
        >
          <svg
            class="category-item-icon"
            fill="blue"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
          </svg>
           Clients
        </li>
      </ul>
    </div>
    <div
      class="menu-labels"
    >
      <p
        class="menu-labels-title"
      >
        Labels
      </p>
      <ul
        class="labels"
      >
        <li
          class="label-item"
        >
          <svg
            class="label-item-icon"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.59 13.41l-7.17 7.17a2 2 0 0 1-2.83 0L2 12V2h10l8.59 8.59a2 2 0 0 1 0 2.82z"
            />
            <line
              x1="7"
              x2="7.01"
              y1="7"
              y2="7"
            />
          </svg>
           Family
        </li>
        <li
          class="label-item"
        >
          <svg
            class="label-item-icon"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.59 13.41l-7.17 7.17a2 2 0 0 1-2.83 0L2 12V2h10l8.59 8.59a2 2 0 0 1 0 2.82z"
            />
            <line
              x1="7"
              x2="7.01"
              y1="7"
              y2="7"
            />
          </svg>
           Work
        </li>
        <li
          class="label-item"
        >
          <svg
            class="label-item-icon"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.59 13.41l-7.17 7.17a2 2 0 0 1-2.83 0L2 12V2h10l8.59 8.59a2 2 0 0 1 0 2.82z"
            />
            <line
              x1="7"
              x2="7.01"
              y1="7"
              y2="7"
            />
          </svg>
           Home
        </li>
        <li
          class="label-item"
        >
          <svg
            class="label-item-icon"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.59 13.41l-7.17 7.17a2 2 0 0 1-2.83 0L2 12V2h10l8.59 8.59a2 2 0 0 1 0 2.82z"
            />
            <line
              x1="7"
              x2="7.01"
              y1="7"
              y2="7"
            />
          </svg>
           Children
        </li>
        <li
          class="label-item"
        >
          <svg
            class="label-item-icon"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.59 13.41l-7.17 7.17a2 2 0 0 1-2.83 0L2 12V2h10l8.59 8.59a2 2 0 0 1 0 2.82z"
            />
            <line
              x1="7"
              x2="7.01"
              y1="7"
              y2="7"
            />
          </svg>
           Holidays
        </li>
        <li
          class="label-item"
        >
          <svg
            class="label-item-icon"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.59 13.41l-7.17 7.17a2 2 0 0 1-2.83 0L2 12V2h10l8.59 8.59a2 2 0 0 1 0 2.82z"
            />
            <line
              x1="7"
              x2="7.01"
              y1="7"
              y2="7"
            />
          </svg>
           Music
        </li>
        <li
          class="label-item"
        >
          <svg
            class="label-item-icon"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.59 13.41l-7.17 7.17a2 2 0 0 1-2.83 0L2 12V2h10l8.59 8.59a2 2 0 0 1 0 2.82z"
            />
            <line
              x1="7"
              x2="7.01"
              y1="7"
              y2="7"
            />
          </svg>
           Photography
        </li>
        <li
          class="label-item"
        >
          <svg
            class="label-item-icon"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.59 13.41l-7.17 7.17a2 2 0 0 1-2.83 0L2 12V2h10l8.59 8.59a2 2 0 0 1 0 2.82z"
            />
            <line
              x1="7"
              x2="7.01"
              y1="7"
              y2="7"
            />
          </svg>
           Film
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
